### PR TITLE
[ORPO] fix orpo chosen-nll loss

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?
Calculate the ORPO chosen nll loss with respect to the chosen completion only rather than the whole prompt+compeletion.

Also return the shifted logits when the model is decoder only

### CODE REVIEW

File: trl/trainer/orpo_trainer.py [hunk 1]
```diff
@@ -774,18 +774,12 @@ def cross_entropy_loss(logits, labels):
...
             return loss
 
-        if self.is_encoder_decoder:
-            labels = concatenated_batch["concatenated_labels"].clone()
-        else:
-            labels = concatenated_batch["concatenated_input_ids"].clone()
-            attention_mask = concatenated_batch["concatenated_attention_mask"]
-            labels = torch.where(attention_mask == 1, labels, self.label_pad_token_id)
-
+        labels = concatenated_batch["concatenated_labels"].clone()
```
Comment (qgallouedec) [inline trl/trainer/orpo_trainer.py:777]: Yes, checked together, if you do 

```python
labels = concatenated_batch["concatenated_input_ids"].clone()
attention_mask = concatenated_batch["concatenated_attention_mask"]
labels = torch.where(attention_mask == 1, labels, self.label_pad_token_id)
```


you don't ignore the prompt.

**Review by qgallouedec — APPROVED** [review]
